### PR TITLE
The Great Radio Rework: NTNET Part 1 of many.

### PR DESCRIPTION
**Pull request by Dejaku51**
## About The Pull Request
port of https://github.com/tgstation/tgstation/pull/54462 so i can port NTNET wiremod components etc

### CODE REVIEW

**Comment by francinum** [comment]
I'll have a look at this but this is liable to be incredibly counterproductive to an aim I've had of *gutting NTNet completely*

**Comment by Dejaku51** [comment]
waiting for review

**Comment by francinum** [comment]
It's more that it's far more complicated than native radio for no justifiable reason.

**Comment by Dejaku51** [comment]
does that mean I'll have to wait a long time before this will be merged?

**Comment by francinum** [comment]
Fuck it, sure, whatever.
I don't have the bandwidth to give a fuck about cleaning up after TG designing themselves into a corner and not using their own tools to claw their way out.

File: code/controllers/subsystem/processing/networks.dm [hunk 1]
```diff
@@ -1,51 +1,523 @@
...
+	return NETWORK_ERROR_OK
+
+
+/datum/controller/subsystem/networks/proc/check_relay_operation(zlevel=0)	//can be expanded later but right now it's true/false.
+	for(var/i in relays)
+		var/obj/machinery/ntnet_relay/n = i
+		if(zlevel && n.z != zlevel)
+			continue
+		if(n.is_operational())
+			return TRUE
```
Comment (Archanial) [inline code/controllers/subsystem/processing/networks.dm]: this should check for virtual z levels

File: code/modules/NTNet/netdata.dm [hunk 2]
```diff
@@ -1,20 +1,101 @@
...
+
+/datum/netlink/Destroy()
+	passkey = null
+	data = null
+	return ..()
+
+// If you don't want to use this fine, but this just shows how the system works
+
+// I hate you all.  I want to operator overload []
+// So fuck you all, Do NOT access data directly you freaks
```
Comment (Archanial) [inline code/modules/NTNet/netdata.dm:32]: 😆

**Review by Archanial — DISMISSED** [review]
Looks fine, still needs testmerge

**Comment by Dejaku51** [comment]
Oh fuck off

**Pull request closed without merging**